mysql/provider.go

```diff
@@ -53,6 +55,12 @@ func Provider() terraform.ResourceProvider {
 				DefaultFunc: schema.EnvDefaultFunc("MYSQL_PASSWORD", nil),
 			},
 
+			"proxy": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringMatch(regexp.MustCompile("^socks5h?://.*:\\d+$"), "The proxy URL is not a valid socks url."),
```

Comment:
Just to make sure, you don't have `DefaultFunc` here because the `ALL_PROXY` and/or `all_proxy` takes care of reading from the environment?

Reply:
Yeah `schema.MultiEnvDefaultFunc` is a good candidate for this scenario.

`because env vars supersede the proxy defined in the tf file right ?` <-- The value defined in the tf file takes precedence so what you have for https://github.com/terraform-providers/terraform-provider-mysql/pull/102/files#diff-12848d8d421c2fa28469f9cfc40602f0R82 is 👍 .  We don't want the env variable(s) value to supersede the defined value in the tf file.

Reply:
Is this change 5076acfe9d90cfe52e9c0612a12a285ae6e4c625 ok for you ?

Reply:
Looks great. Thanks!

Reply:
@barryib, @davidji99: Maybe the both of you could have a look at this PR: https://github.com/terraform-providers/terraform-provider-mysql/pull/108

Somehow the code ended up in a way that leads to the proxy being set twice (as proxy and forwarder). My PR fixes this. 

Not sure if removing proxy.FromEnvironment() instead is desired.